blockchain/types/tx_internal_data_access_list.go

```diff
@@ -0,0 +1,494 @@
+		Amount:       new(big.Int),
+		GasLimit:     0,
+		AccessList:   AccessList{},
+		Price:        new(big.Int),
+		V:            new(big.Int),
+		R:            new(big.Int),
+		S:            new(big.Int),
+		ChainID:      new(big.Int),
+	}
+}
+
+func newTxInternalDataAccessListWithValues(nonce uint64, to *common.Address, amount *big.Int, gasLimit uint64, gasPrice *big.Int, data []byte, accessList AccessList, chainID *big.Int) *TxInternalDataAccessList {
+	d := newTxInternalDataAccessList()
+
+	d.AccountNonce = nonce
+	d.Recipient = to
+	d.GasLimit = gasLimit
+
+	if len(data) > 0 {
+		d.Payload = common.CopyBytes(data)
+	}
+	if amount != nil {
+		d.Amount.Set(amount)
+	}
+	if gasPrice != nil {
+		d.Price.Set(gasPrice)
+	}
+
+	if accessList != nil {
+		copy(d.AccessList, accessList)
+	}
+
+	if chainID != nil {
+		d.ChainID.Set(chainID)
+	}
+
+	return d
+}
+
+func newTxInternalDataAccessListWithMap(values map[TxValueKeyType]interface{}) (*TxInternalDataAccessList, error) {
+	d := newTxInternalDataAccessList()
+
+	if v, ok := values[TxValueKeyNonce].(uint64); ok {
+		d.AccountNonce = v
+		delete(values, TxValueKeyNonce)
+	} else {
+		return nil, errValueKeyNonceMustUint64
+	}
+
+	if v, ok := values[TxValueKeyTo].(common.Address); ok {
+		d.Recipient = &v
+		delete(values, TxValueKeyTo)
+	} else {
+		return nil, errValueKeyToMustAddress
+	}
+
+	if v, ok := values[TxValueKeyAmount].(*big.Int); ok {
+		d.Amount.Set(v)
+		delete(values, TxValueKeyAmount)
+	} else {
+		return nil, errValueKeyAmountMustBigInt
+	}
+
+	if v, ok := values[TxValueKeyData].([]byte); ok {
+		d.Payload = common.CopyBytes(v)
+		delete(values, TxValueKeyData)
+	} else {
+		return nil, errValueKeyDataMustByteSlice
+	}
+
+	if v, ok := values[TxValueKeyGasLimit].(uint64); ok {
+		d.GasLimit = v
+		delete(values, TxValueKeyGasLimit)
+	} else {
+		return nil, errValueKeyGasLimitMustUint64
+	}
+
+	if v, ok := values[TxValueKeyGasPrice].(*big.Int); ok {
+		d.Price.Set(v)
+		delete(values, TxValueKeyGasPrice)
+	} else {
+		return nil, errValueKeyGasPriceMustBigInt
+	}
+
+	if v, ok := values[TxValueKeyAccessList].(AccessList); ok {
+		d.AccessList = make(AccessList, len(v))
+		copy(d.AccessList, v)
+		delete(values, TxValueKeyAccessList)
+	} else {
+		return nil, errValueKeyAccessListInvalid
+	}
+
+	if v, ok := values[TxValueKeyChainID].(*big.Int); ok {
+		d.ChainID.Set(v)
+		delete(values, TxValueKeyChainID)
+	} else {
+		return nil, errValueKeyChainIDInvalid
+	}
+
+	if len(values) != 0 {
+		for k := range values {
+			logger.Warn("unnecessary key", k.String())
+		}
+		return nil, errUndefinedKeyRemains
+	}
+
+	return d, nil
+}
+
+func (t *TxInternalDataAccessList) Type() TxType {
+	return TxTypeAccessList
+}
+
+func (t *TxInternalDataAccessList) GetRoleTypeForValidation() accountkey.RoleType {
+	return accountkey.RoleTransaction
+}
+
+func (t *TxInternalDataAccessList) ChainId() *big.Int {
+	if t.ChainID != nil {
+		return t.ChainID
+	}
+	return deriveChainId(t.V)
```

Comment:
I think `deriveChainId(t.V)` returns invalid ChainID for TxInternalDataAccessList. How about just returning `t.ChainID`? I could return nil, but panic is better than inconsistency for this case.